File: frontend/src/types/mediaItem.ts
export interface LibrarySection {
	ID: string;
	Type: string; // "movie" or "show"
	Title: string;
	TotalSize: number;
	MediaItems: MediaItem[];
}

export interface MediaItem {
	RatingKey: string;
	LibraryTitle: string;
	Type: string; // "movie" or "show"
	Title: string;
	Year: number;
	ExistInDatabase: boolean;
	Thumb?: string;
	ContentRating?: string;
	Summary?: string;
	UpdatedAt?: number;
	Guids: Guid[];
	Movie?: MediaItemMovie;
	Series?: MediaItemSeries;
}

export interface Guid {
	Provider?: string;
	ID?: string;
	Rating?: string;
}

export interface MediaItemMovie {
	File: MediaItemFile;
}

export interface MediaItemSeries {
	Seasons: MediaItemSeason[];
	SeasonCount: number;
	EpisodeCount: number;
}

export interface MediaItemSeason {
	RatingKey: string;
	SeasonNumber: number;
	Title: string;
	Episodes: MediaItemEpisode[];
}

export interface MediaItemEpisode {
	RatingKey: string;
	Title: string;
	SeasonNumber: number;
	EpisodeNumber: number;
	File: File;
}

export interface MediaItemFile {
	Path: string;
	Size: number;
	Duration: number;
}

File: backend/internal/modals/MediaItem.go
package modals

type LibrarySection struct {
	ID         string      `json:"ID"`
	Type       string      `json:"Type"` // "movie" or "show"
	Title      string      `json:"Title"`
	TotalSize  int         `json:"TotalSize"`
	MediaItems []MediaItem `json:"MediaItems"`
}

type MediaItem struct {
	RatingKey      string  `json:"RatingKey"`
	Type           string  `json:"Type"` // "movie" or "series"
	LibraryTitle   string  `json:"LibraryTitle"`
	Title          string  `json:"Title"`
	Year           int     `json:"Year"`
	Thumb          string  `json:"Thumb,omitempty"`
	AudienceRating float64 `json:"AudienceRating,omitempty"`
	UserRating     float64 `json:"UserRating,omitempty"`
	ContentRating  string  `json:"ContentRating,omitempty"`
	Summary        string  `json:"Summary,omitempty"`
	UpdatedAt      int64   `json:"UpdatedAt,omitempty"`
	Guids          []Guid  `json:"Guids,omitempty"`

	Movie  *MediaItemMovie  `json:"Movie,omitempty"`
	Series *MediaItemSeries `json:"Series,omitempty"`
}

type Guid struct {
	Provider string `json:"Provider"`
	ID       string `json:"ID"`
}

type MediaItemMovie struct {
	File MediaItemFile `json:"File"`
}

type MediaItemSeries struct {
	Seasons      []MediaItemSeason `json:"Seasons"`
	SeasonCount  int               `json:"SeasonCount"`
	EpisodeCount int               `json:"EpisodeCount"`
	Location     string            `json:"Location"`
}

type MediaItemSeason struct {
	RatingKey    string             `json:"RatingKey"`
	SeasonNumber int                `json:"SeasonNumber"`
	Title        string             `json:"Title"`
	Episodes     []MediaItemEpisode `json:"Episodes"`
}

type MediaItemEpisode struct {
	RatingKey     string        `json:"RatingKey"`
	Title         string        `json:"Title"`
	SeasonNumber  int           `json:"SeasonNumber"`
	EpisodeNumber int           `json:"EpisodeNumber"`
	File          MediaItemFile `json:"File"`
}

type MediaItemFile struct {
	Path     string `json:"Path"`
	Size     int64  `json:"Size"`
	Duration int64  `json:"Duration"`
}
